database/tests/audit_test.go

```
@@ -0,0 +1,109 @@
+package tests
+
+import (
+	"fmt"
+	"path"
+	"sync"
+	"testing"
+
+	"github.com/thrasher-/gocryptotrader/database/repository/audit"
+	auditPSQL "github.com/thrasher-/gocryptotrader/database/repository/audit/postgres"
+	auditSQlite "github.com/thrasher-/gocryptotrader/database/repository/audit/sqlite"
+
+	"github.com/thrasher-/gocryptotrader/database"
+	"github.com/thrasher-/gocryptotrader/database/drivers"
+	dbpsql "github.com/thrasher-/gocryptotrader/database/drivers/postgres"
+	dbsqlite "github.com/thrasher-/gocryptotrader/database/drivers/sqlite"
+)
+
+func TestAudit(t *testing.T) {
+	testCases := []struct {
+		name   string
+		config database.Config
+		setup  func() error
+		audit  audit.Repository
+		runner func(t *testing.T)
+		output interface{}
+	}{
+		{
+			"SQLite",
+			database.Config{
+				Driver:            "sqlite",
+				ConnectionDetails: drivers.ConnectionDetails{Database: path.Join(tempDir, "./testdb.db")},
+			},
+			dbsqlite.Setup,
+			auditSQlite.Audit(),
+			writeAudit,
+			nil,
+		},
+		{
+			"Postgres",
+			postgresTestDatabase,
+			dbpsql.Setup,
+			auditPSQL.Audit(),
+			writeAudit,
+			nil,
+		},
+	}
+
+	for _, tests := range testCases {
+		test := tests
+
+		t.Run(test.name, func(t *testing.T) {
+
+			if !checkValidConfig(t, &test.config.ConnectionDetails) {
+				t.Skip("database not configured skipping test")
+			}
+
+			dbConn, err := connectToDatabase(t, &test.config)
```

This error is not being checked